### PR TITLE
Addition of op-amp section to labs, content added to empty section on analog inputs

### PR DESCRIPTION
**Pull request by jcwilliamson**
Addition of a section giving  an overview of the functioning of operational amplifiers, and a comparator circuit that may be useful for teams to know. Also includes an illustration of the operational amplifier circuit.

### CODE REVIEW

**Comment by jcwilliamson** [comment]
Op amp in comparator was in inverting configuration, Changed to non inverting

**Comment by kierdavis** [comment]
Just had a look through, looks great so far! My only criticism is that it might be a bit unclear from the comparator diagram what someone should do with "Vin" when building the circuit, or what the other voltage supply is doing. Given that ultimately the students will be using identical PSUs for the supply and input voltages, I'd suggest using the same circuit symbol and clearly labelling which is which.

**Comment by jcwilliamson** [comment]
Thanks for your feedback Kier, I'll explain what to do with Vin in my next commit. Would you mind clarifying for me what you mean when you say "the other voltage supply"?

**Comment by kierdavis** [comment]
Cool :)

> Would you mind clarifying for me what you mean when you say "the other voltage supply"?

Sorry for the vagueness; I was referring to the component labelled V1.

File: electronics/sections/opamps.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Use of Operational Amplifiers (Op Amps)
+
+An operational amplifier is a device that takes 2 inputs, measures the difference between them, then outputs this voltage increased by a factor known as "gain". The gain of an operational amplifier is the extent to which the difference in voltage between its two inputs is increased. The two inputs of are normally identified as the "inverting" and "non-inverting", indicated on the schematic by "-" and "+" respectively.
+An op-amp also requires two different supply voltages to function. Op-amps can be characterised by the power supplies they require, which divides them into 2 groups, single supply, which require a positive supply and ground, and dual supply, which require a positive and negative supply, along with a ground connection. These supplies determine the output range of an op-amp, as it will be unable to exceed these voltages, regardless of its gain. The gain of an op-amp is typically on the order of 100,000. In other words, if the supply were large enough, a 1 mV difference between the inputs would result in a 100V output.
```
Comment (trickeydan) [inline electronics/sections/opamps.md]: If you'd like these to render as separate paragraphs, you'll need an extra line break here. :)

File: electronics/sections/opamps.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+In an analogue circuit, a gain of this magnitude is generally not useful, as the op-amp will saturate for extremely small signals. In an ideal op-amp, saturation means that the output is equal to the supply voltage. However, in the case of a real op-amp, this is normally slightly lower than the supply voltage. A common solution to this problem is to build a "feedback loop". This is where the output of the op-amp is connected back to one of the inputs through a resistor. Several configurations of this are possible, however, in all cases the feedback is used to reduce the gain.
+
+However, In circuits where analogue and digital signals are both present, the saturation of an "open loop" op-amp can sometimes be useful. An "open loop" configuration has no connection between its input and its output. This is the condition under which an op-amps gain is usually measured.
+
+\begin{center} \includegraphics[height=5cm] {img/comparator.png} \end{center}
+
+Try building the circuit shown above, connecting the signal generator in place of voltage supply V2. Set the signal generator to produce a sinusoidal output with a peak to peak amplitude of 5V and a DC offset of 2.5 volts.
```
Comment (trickeydan) [inline electronics/sections/opamps.md]:
```suggestion
5V and a DC offset of 2.5V. 
```

or

```suggestion
5 volts and a DC offset of 2.5 volts. 
```

Comment (trickeydan) [inline electronics/sections/opamps.md]:
```suggestion
Try building the circuit shown above, connecting the signal generator in place of voltage supply V2. Set the signal generator to produce a sinusoidal output with a peak to peak amplitude of 5 volts and a DC offset of 2.5 volts. 
```
or
```suggestion
Try building the circuit shown above, connecting the signal generator in place of voltage supply V2. Set the signal generator to produce a sinusoidal output with a peak to peak amplitude of 5V and a DC offset of 2.5V. 
```

File: electronics/sections/opamps.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#Use of Operational Amplifiers (Op Amps)
+
+An operational amplifier is a device that takes 2 inputs, measures the difference between them, then outputs this voltage increased
+by a factor known as "gain". The gain of an operational amplifier is the extent to which the difference in voltage between its two inputs
+is increased. The two inputs of are normally identified as the "inverting" and "non-inverting", indicated on the schematic by "-" and "+" respectively.
+An op-am also requires two different supply voltages to function. These supplies determine the output range of an op-amp, as it will be unable to exceed these voltages, regardless of its gain. The gain of an op-amp is typically
```
Comment (Dasepho) [inline electronics/sections/opamps.md]: typo: op-amp

File: electronics/sections/analogue.md [hunk 4]
```diff
@@ -1 +1,32 @@
...
+
+$$\text{Resolution} = \frac{5}{1024}$$
+
+This gives the resolution of the ADC as 4.9mV. This function of the Arduino can be accessed using the **analogread(analogpin)** function on anything connected to the Arduino's dedicated analogue pins. Your Arduino has 6 such pins, noted by A0, A1, A2, A3,A4 ,A5. This may be useful for some of your sensors, as some of them do not return a digital value, and instead return a range of analogue values, particularly if you need more detail from the sensors than simply on or off.
+
+## Potential Dividers
+A potential divider is a digital circuit that consists of 2 or more resistors. This is useful for bringing a voltage that is out of range of your Arduino down to a lower voltage which is in range. Here, Ohm's law causes a drop in voltage across the first resistor (R1) which is proportional to its resistance, This leaves the second resistor (R2) to drop the remaining voltage.
```
Comment (Dasepho) [inline electronics/sections/analogue.md]: Digital? A potential divider can output an analogue voltage, e.g. potentiometer?

Comment (jcwilliamson) [reply electronics/sections/analogue.md]: Good spot there Dan. I have no idea what I was even attempting to write, I think it may have just been late.

**Comment by trickeydan** [comment]
I'm merging this so that we can work on it on the org.

**Pull request merged**